docs/modules/ROOT/pages/configure-cmos.adoc

```diff
@@ -0,0 +1,229 @@
+
+- `CLUSTER_MONITOR_USER=${CLUSTER_MONITOR_USER:-admin}`: The username of couchbase cluster to monitor.
+- `CLUSTER_MONITOR_PWD=${CLUSTER_MONITOR_PWD:-password}`: The password of couchbase cluster to monitor.
+- `CLUSTER_MONITOR_ENDPOINT=${CLUSTER_MONITOR_ENDPOINT:-http://localhost:7196}`: The endpoint of couchbase cluster to monitor.
+
+The cluster manager  exposes a REST API endpoint from the container so that it can be used externally to add/remove Couchbase clusters. 
+The cluster monitor is also capable of running any scripts found in `/etc/healthcheck/` folder to perform health checks on the Couchbase cluster.
+
+== Prometheus
+
+Prometheus has various configuration options exposed to the user, almost entirely using files.
+You can mount these configuration files to the CMOS container using https://docs.docker.com/storage/volumes/[docker volumes^] or https://docs.docker.com/storage/bind-mounts/[bind mounts^].
+
+[NOTE]
+To pick up any configuration changes after the prometheus service has started, you may need to reload the config files using the `reload` endpoint.
+
+Below are the environment variables used for Prometheus configuration.
+You can further read on prometheus configuration on the https://prometheus.io/docs/prometheus/latest/configuration/configuration/[official documentation website^].
+
+[console]
+----
+# Prometheus configuration file
+PROMETHEUS_CONFIG_FILE=${PROMETHEUS_CONFIG_FILE:-/etc/prometheus/prometheus-runtime.yml}
+
+# Prometheus configuration file template
+PROMETHEUS_CONFIG_TEMPLATE_FILE=${PROMETHEUS_CONFIG_TEMPLATE_FILE:-/etc/prometheus/prometheus-template.yml}
+
+# Prometheus url sub path
+PROMETHEUS_URL_SUBPATH=${PROMETHEUS_URL_SUBPATH-/prometheus/}
+
+# Prometheus storage path
+PROMETHEUS_STORAGE_PATH=${PROMETHEUS_STORAGE_PATH-/prometheus}
+
+# Microlith configuration
+PROMETHEUS_DYNAMIC_INTERNAL_DIR=${PROMETHEUS_DYNAMIC_INTERNAL_DIR:-/etc/prometheus/couchbase/monitoring/}
+----
+
+=== Endpoints
+
+In CMOS, a https://prometheus.io/docs/prometheus/latest/configuration/configuration/#file_sd_config[file-based service discovery^] approach is used to support adding/removing endpoints.
+These endpoints are used to dynamically scrape metrics.
+
+Based on which services are enabled, the microlith will construct a set of dynamic endpoints.
+These endpoints are created for each service by adding a JSON file to the `${PROMETHEUS_DYNAMIC_INTERNAL_DIR:-/etc/prometheus/couchbase/monitoring/}`.
+
+To add Couchbase Server endpoints, create a similar JSON format file like the example in the `/etc/prometheus/couchbase/custom/` directory mounted on the container.
+In order to always stay current with the system state, this file is periodically rescanned to add or remove targets.
+
+Whereas, this enables addition of a new scrape target, it does not automatically add dashboards to Grafana for this scrape target and you will have to do that manually.
```

Comment:
Wasn't clear on the meaning here, so correct me if my interpretation is wrong, but maybe something like:

```suggestion
While this enables the addition of a new scrape target, it will not automatically add dashboards to Grafana. The dashboards will need to be added manually for the relevant scrape target.
```